[Automation] java run fails for io.circe:circe-core_3:0.14.11

## java run on ubuntu-latest / JDK 25 / mode current-defaults

Failure kind: java run
OS: ubuntu-latest
JDK: 25
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="0.14.11" ./gradlew clean javaTest -Pcoordinates="io.circe:circe-core_3:0.14.10"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25496563122/job/74818140142
Last 300 lines of the log:
```console
Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 7s
9 actionable tasks: 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :tckTest
====================
Testing library: io.circe:circe-core_3:0.14.10
Command: `/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/gradlew nativeTest`
Executing test...
-------
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :compileScala NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :jar
> Task :checkTestTimeoutAnnotations
> Task :processTestResources NO-SOURCE
> Task :tck-test-support:compileJava
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes
> Task :compileTestJava NO-SOURCE
> Task :tck-test-support:jar

> Task :compileTestScala
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.circe/circe-core_3/0.14.10/src/test/scala/io_circe/circe_core_3/Circe_core_3Test.scala:64:92: [33mclass[0m [35mDecoderNotDeriveSum[0m cannot be accessed as a member of io.circe.derivation.type from [33mobject[0m [35mDerivedJob[0m.
  private[circe] [33mclass[0m [35mDecoderNotDeriveSum[0m can only be accessed from [33mpackage[0m [35mio.circe[0m in [33mpackage[0m [35mio[0m.
one error found

> Task :compileTestScala FAILED


FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestScala'.
> [Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.circe/circe-core_3/0.14.10/build/reports/problems/problems-report.html
Compilation failed


* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).
Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.


BUILD FAILED in 21s
For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
13 actionable tasks: 5 executed, 8 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for io.circe:circe-core_3:0.14.10 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 30s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 0.14.11 at 2026-05-07T12:49:17Z with exit code 1 after 31s
❌ ERROR: 'native-image run' for 0.14.11 failed with exit code 1.
SOFT-FAIL[native-image run][0.14.11][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="0.14.11" ./gradlew clean test -Pcoordinates="io.circe:circe-core_3:0.14.10"]
========================================================================================
 io.circe:circe-core_3:0.14.10:0.14.11 stage javac compile
========================================================================================
Starting stage 'javac compile' for 0.14.11 at 2026-05-07T12:49:17Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="0.14.11" ./gradlew clean compileTestJava -Pcoordinates="io.circe:circe-core_3:0.14.10"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.circe/circe-core_3/0.14.10/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 965ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :compileScala NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :compileTestJava NO-SOURCE

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.circe/circe-core_3/0.14.10/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 902ms
9 actionable tasks: 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 3s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 0.14.11 at 2026-05-07T12:49:20Z with exit code 0 after 3s
========================================================================================
 io.circe:circe-core_3:0.14.10:0.14.11 stage java run
========================================================================================
Starting stage 'java run' for 0.14.11 at 2026-05-07T12:49:20Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="0.14.11" ./gradlew clean javaTest -Pcoordinates="io.circe:circe-core_3:0.14.10"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.circe/circe-core_3/0.14.10/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 908ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :javaTest
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes UP-TO-DATE
> Task :checkTestTimeoutAnnotations
> Task :tck-test-support:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :compileScala NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :compileTestJava NO-SOURCE

> Task :compileTestScala
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.circe/circe-core_3/0.14.10/src/test/scala/io_circe/circe_core_3/Circe_core_3Test.scala:64:92: [33mclass[0m [35mDecoderNotDeriveSum[0m cannot be accessed as a member of io.circe.derivation.type from [33mobject[0m [35mDerivedJob[0m.
  private[circe] [33mclass[0m [35mDecoderNotDeriveSum[0m can only be accessed from [33mpackage[0m [35mio.circe[0m in [33mpackage[0m [35mio[0m.
one error found

> Task :compileTestScala FAILED


FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestScala'.
> Compilation failed

* Try:
[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.circe/circe-core_3/0.14.10/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
12 actionable tasks: 2 executed, 10 up-to-date
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 7s

> Task :javaTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':javaTest'.
> Java tests failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 9s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'java run' for 0.14.11 at 2026-05-07T12:49:29Z with exit code 1 after 9s
❌ ERROR: 'java run' for 0.14.11 failed with exit code 1.
FAILED[java run][0.14.11][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="0.14.11" ./gradlew clean javaTest -Pcoordinates="io.circe:circe-core_3:0.14.10"]
```

## java run on ubuntu-latest / JDK latest-ea / mode current-defaults

Failure kind: java run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="0.14.11" ./gradlew clean javaTest -Pcoordinates="io.circe:circe-core_3:0.14.10"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25496563122/job/74818140146
Last 300 lines of the log:
```console
Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 7s
9 actionable tasks: 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :tckTest
====================
Testing library: io.circe:circe-core_3:0.14.10
Command: `/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/gradlew nativeTest`
Executing test...
-------
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :compileScala NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :jar
> Task :checkTestTimeoutAnnotations
> Task :processTestResources NO-SOURCE
> Task :tck-test-support:compileJava
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes
> Task :tck-test-support:jar
> Task :compileTestJava NO-SOURCE

> Task :compileTestScala
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.circe/circe-core_3/0.14.10/src/test/scala/io_circe/circe_core_3/Circe_core_3Test.scala:64:92: [33mclass[0m [35mDecoderNotDeriveSum[0m cannot be accessed as a member of io.circe.derivation.type from [33mobject[0m [35mDerivedJob[0m.
  private[circe] [33mclass[0m [35mDecoderNotDeriveSum[0m can only be accessed from [33mpackage[0m [35mio.circe[0m in [33mpackage[0m [35mio[0m.
one error found

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestScala'.
> Compilation failed

* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).

> Task :compileTestScala FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.circe/circe-core_3/0.14.10/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
13 actionable tasks: 5 executed, 8 up-to-date

BUILD FAILED in 18s

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for io.circe:circe-core_3:0.14.10 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 27s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 0.14.11 at 2026-05-07T12:49:09Z with exit code 1 after 28s
❌ ERROR: 'native-image run' for 0.14.11 failed with exit code 1.
SOFT-FAIL[native-image run][0.14.11][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="0.14.11" ./gradlew clean test -Pcoordinates="io.circe:circe-core_3:0.14.10"]
========================================================================================
 io.circe:circe-core_3:0.14.10:0.14.11 stage javac compile
========================================================================================
Starting stage 'javac compile' for 0.14.11 at 2026-05-07T12:49:09Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="0.14.11" ./gradlew clean compileTestJava -Pcoordinates="io.circe:circe-core_3:0.14.10"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.circe/circe-core_3/0.14.10/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 870ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :compileScala NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :compileTestJava NO-SOURCE

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.circe/circe-core_3/0.14.10/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 865ms
9 actionable tasks: 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 2s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 0.14.11 at 2026-05-07T12:49:12Z with exit code 0 after 3s
========================================================================================
 io.circe:circe-core_3:0.14.10:0.14.11 stage java run
========================================================================================
Starting stage 'java run' for 0.14.11 at 2026-05-07T12:49:12Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="0.14.11" ./gradlew clean javaTest -Pcoordinates="io.circe:circe-core_3:0.14.10"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.circe/circe-core_3/0.14.10/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 801ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :javaTest
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :checkTestTimeoutAnnotations
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :tck-test-support:processResources NO-SOURCE
> Task :compileScala NO-SOURCE
> Task :tck-test-support:classes UP-TO-DATE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :tck-test-support:jar UP-TO-DATE
> Task :compileTestJava NO-SOURCE

> Task :compileTestScala
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.circe/circe-core_3/0.14.10/src/test/scala/io_circe/circe_core_3/Circe_core_3Test.scala:64:92: [33mclass[0m [35mDecoderNotDeriveSum[0m cannot be accessed as a member of io.circe.derivation.type from [33mobject[0m [35mDerivedJob[0m.
  private[circe] [33mclass[0m [35mDecoderNotDeriveSum[0m can only be accessed from [33mpackage[0m [35mio.circe[0m in [33mpackage[0m [35mio[0m.
one error found

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestScala'.
> Compilation failed

* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).

> Task :compileTestScala FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.circe/circe-core_3/0.14.10/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 6s
12 actionable tasks: 2 executed, 10 up-to-date

> Task :javaTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':javaTest'.
> Java tests failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 8s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'java run' for 0.14.11 at 2026-05-07T12:49:21Z with exit code 1 after 9s
❌ ERROR: 'java run' for 0.14.11 failed with exit code 1.
FAILED[java run][0.14.11][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="0.14.11" ./gradlew clean javaTest -Pcoordinates="io.circe:circe-core_3:0.14.10"]
```

## java run on ubuntu-latest / JDK latest-ea / mode future-defaults-all

Failure kind: java run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: future-defaults-all
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="0.14.11" ./gradlew clean javaTest -Pcoordinates="io.circe:circe-core_3:0.14.10"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25496563122/job/74818140131
Last 300 lines of the log:
```console

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 7s
9 actionable tasks: 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :tckTest
====================
Testing library: io.circe:circe-core_3:0.14.10
Command: `/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/gradlew nativeTest`
Executing test...
-------
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :compileScala NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :jar
> Task :checkTestTimeoutAnnotations
> Task :processTestResources NO-SOURCE
> Task :tck-test-support:compileJava
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes
> Task :tck-test-support:jar
> Task :compileTestJava NO-SOURCE

> Task :compileTestScala
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.circe/circe-core_3/0.14.10/src/test/scala/io_circe/circe_core_3/Circe_core_3Test.scala:64:92: [33mclass[0m [35mDecoderNotDeriveSum[0m cannot be accessed as a member of io.circe.derivation.type from [33mobject[0m [35mDerivedJob[0m.
  private[circe] [33mclass[0m [35mDecoderNotDeriveSum[0m can only be accessed from [33mpackage[0m [35mio.circe[0m in [33mpackage[0m [35mio[0m.
one error found


FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestScala'.
> Compilation failed

* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Task :compileTestScala FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.circe/circe-core_3/0.14.10/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
13 actionable tasks: 5 executed, 8 up-to-date

BUILD FAILED in 18s

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for io.circe:circe-core_3:0.14.10 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 28s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 0.14.11 at 2026-05-07T12:49:12Z with exit code 1 after 28s
❌ ERROR: 'native-image run' for 0.14.11 failed with exit code 1.
SOFT-FAIL[native-image run][0.14.11][GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="0.14.11" ./gradlew clean test -Pcoordinates="io.circe:circe-core_3:0.14.10"]
========================================================================================
 io.circe:circe-core_3:0.14.10:0.14.11 stage javac compile
========================================================================================
Starting stage 'javac compile' for 0.14.11 at 2026-05-07T12:49:12Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="0.14.11" ./gradlew clean compileTestJava -Pcoordinates="io.circe:circe-core_3:0.14.10"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.circe/circe-core_3/0.14.10/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 925ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :compileScala NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :compileTestJava NO-SOURCE

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.circe/circe-core_3/0.14.10/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 941ms
9 actionable tasks: 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 2s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 0.14.11 at 2026-05-07T12:49:15Z with exit code 0 after 3s
========================================================================================
 io.circe:circe-core_3:0.14.10:0.14.11 stage java run
========================================================================================
Starting stage 'java run' for 0.14.11 at 2026-05-07T12:49:15Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="0.14.11" ./gradlew clean javaTest -Pcoordinates="io.circe:circe-core_3:0.14.10"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.circe/circe-core_3/0.14.10/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 922ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :javaTest
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :checkTestTimeoutAnnotations
> Task :compileJava NO-SOURCE
> Task :compileScala NO-SOURCE
> Task :processResources NO-SOURCE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes UP-TO-DATE
> Task :classes UP-TO-DATE
> Task :processTestResources NO-SOURCE
> Task :tck-test-support:jar UP-TO-DATE
> Task :compileTestJava NO-SOURCE

> Task :compileTestScala
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.circe/circe-core_3/0.14.10/src/test/scala/io_circe/circe_core_3/Circe_core_3Test.scala:64:92: [33mclass[0m [35mDecoderNotDeriveSum[0m cannot be accessed as a member of io.circe.derivation.type from [33mobject[0m [35mDerivedJob[0m.
  private[circe] [33mclass[0m [35mDecoderNotDeriveSum[0m can only be accessed from [33mpackage[0m [35mio.circe[0m in [33mpackage[0m [35mio[0m.
one error found

> Task :compileTestScala FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.circe/circe-core_3/0.14.10/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
12 actionable tasks: 2 executed, 10 up-to-date

* What went wrong:
Execution failed for task ':compileTestScala'.
> Compilation failed

* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 7s

> Task :javaTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':javaTest'.
> Java tests failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 9s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'java run' for 0.14.11 at 2026-05-07T12:49:25Z with exit code 1 after 10s
❌ ERROR: 'java run' for 0.14.11 failed with exit code 1.
FAILED[java run][0.14.11][GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="0.14.11" ./gradlew clean javaTest -Pcoordinates="io.circe:circe-core_3:0.14.10"]
```

--- a/metadata/io.circe/circe-core_3/0.14.11/reachability-metadata.json
+++ b/metadata/io.circe/circe-core_3/0.14.11/reachability-metadata.json
@@ -1,0 +1,16 @@
+{
+  "reflection": [
+    {
+      "condition": {
+        "typeReached": "io.circe.JsonNumber$"
+      },
+      "type": "io.circe.Json$JNumber[]"
+    },
+    {
+      "condition": {
+        "typeReached": "io.circe.Decoder$$anon$58"
+      },
+      "type": "java.time.format.DateTimeFormatterBuilder$DateTimePrinterParser[]"
+    }
+  ]
+}

--- a/metadata/io.circe/circe-core_3/index.json
+++ b/metadata/io.circe/circe-core_3/index.json
@@ -1,6 +1,19 @@
 [
   {
     "latest" : true,
+    "metadata-version" : "0.14.11",
+    "language" : {
+      "name" : "scala",
+      "version" : "3"
+    },
+    "tested-versions" : [
+      "0.14.11"
+    ],
+    "allowed-packages" : [
+      "io.circe"
+    ]
+  },
+  {
     "metadata-version" : "0.14.10",
     "source-code-url" : "https://repo.maven.apache.org/maven2/io/circe/circe-core_3/$version$/circe-core_3-$version$-sources.jar",
     "repository-url" : "https://github.com/circe/circe",

--- a/metadata/io.circe/circe-core_3/index.json
+++ b/metadata/io.circe/circe-core_3/index.json
@@ -6,6 +6,11 @@
       "name" : "scala",
       "version" : "3"
     },
+    "source-code-url" : "https://repo.maven.apache.org/maven2/io/circe/circe-core_3/$version$/circe-core_3-$version$-sources.jar",
+    "repository-url" : "https://github.com/circe/circe",
+    "test-code-url" : "https://github.com/circe/circe/tree/v$version$/modules/tests/shared/src/test",
+    "documentation-url" : "https://repo.maven.apache.org/maven2/io/circe/circe-core_3/$version$/circe-core_3-$version$-javadoc.jar",
+    "description" : "Circe Core provides the core JSON data model and type classes for encoding and decoding JSON in Scala. It includes the JSON AST, cursor APIs, codec abstractions, printers, parser interfaces, and standard instances used by other Circe modules.",
     "tested-versions" : [
       "0.14.11"
     ],

--- a/stats/io.circe/circe-core_3/0.14.11/execution-metrics.json
+++ b/stats/io.circe/circe-core_3/0.14.11/execution-metrics.json
@@ -1,0 +1,87 @@
+{
+  "fix_java_run_fail:2026-05-07": {
+    "timestamp": "2026-05-07T20:52:33.102505Z",
+    "library": "io.circe:circe-core_3:0.14.11",
+    "starting_commit": "7f68cfb3fa233b237be2eba9e98850af6316fbdc",
+    "ending_commit": "fa27fa569f23f12bc918a24888a4dbb4c9a61acc",
+    "previous_library": "io.circe:circe-core_3:0.14.10",
+    "strategy_name": "java_run_iterative_with_coverage_sources_pi_gpt-5.5",
+    "agent": "pi",
+    "model": "oca/gpt-5.5",
+    "status": "success",
+    "stats": {
+      "version": "0.14.11",
+      "dynamicAccess": {
+        "breakdown": {},
+        "coverageRatio": 1.0,
+        "coveredCalls": 0,
+        "totalCalls": 0
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 5854,
+          "missed": 173258,
+          "ratio": 0.032683,
+          "total": 179112
+        },
+        "line": {
+          "covered": 856,
+          "missed": 3629,
+          "ratio": 0.190858,
+          "total": 4485
+        },
+        "method": {
+          "covered": 554,
+          "missed": 5485,
+          "ratio": 0.091737,
+          "total": 6039
+        }
+      }
+    },
+    "previous_library_stats": {
+      "version": "0.14.10",
+      "dynamicAccess": {
+        "breakdown": {},
+        "coverageRatio": 1.0,
+        "coveredCalls": 0,
+        "totalCalls": 0
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 6863,
+          "missed": 173089,
+          "ratio": 0.038138,
+          "total": 179952
+        },
+        "line": {
+          "covered": 968,
+          "missed": 3555,
+          "ratio": 0.214017,
+          "total": 4523
+        },
+        "method": {
+          "covered": 623,
+          "missed": 5404,
+          "ratio": 0.103368,
+          "total": 6027
+        }
+      }
+    },
+    "metrics": {
+      "input_tokens_used": 56527,
+      "cached_input_tokens_used": 640512,
+      "output_tokens_used": 6558,
+      "iterations": 1,
+      "cost_usd": 0.517,
+      "tested_library_loc": 856,
+      "metadata_entries": 2,
+      "previous_library_metadata_entries": 0,
+      "code_coverage_percent": 19.09,
+      "previous_library_coverage_percent": 21.4
+    },
+    "artifacts": {
+      "test_file": "tests/src/io.circe/circe-core_3/0.14.11/src/test/scala/io_circe/circe_core_3/Circe_core_3Test.scala",
+      "metadata_file": "metadata/io.circe/circe-core_3/0.14.11/reachability-metadata.json"
+    }
+  }
+}

--- a/stats/io.circe/circe-core_3/0.14.11/stats.json
+++ b/stats/io.circe/circe-core_3/0.14.11/stats.json
@@ -1,0 +1,31 @@
+{
+  "versions" : [ {
+    "version" : "0.14.11",
+    "dynamicAccess" : {
+      "breakdown" : { },
+      "coverageRatio" : 1.0,
+      "coveredCalls" : 0,
+      "totalCalls" : 0
+    },
+    "libraryCoverage" : {
+      "instruction" : {
+        "covered" : 5854,
+        "missed" : 173258,
+        "ratio" : 0.032683,
+        "total" : 179112
+      },
+      "line" : {
+        "covered" : 856,
+        "missed" : 3629,
+        "ratio" : 0.190858,
+        "total" : 4485
+      },
+      "method" : {
+        "covered" : 554,
+        "missed" : 5485,
+        "ratio" : 0.091737,
+        "total" : 6039
+      }
+    }
+  } ]
+}

--- a/tests/src/io.circe/circe-core_3/0.14.11/.gitignore
+++ b/tests/src/io.circe/circe-core_3/0.14.11/.gitignore
@@ -1,0 +1,4 @@
+gradlew.bat
+gradlew
+gradle/
+build/

--- a/tests/src/io.circe/circe-core_3/0.14.11/build.gradle
+++ b/tests/src/io.circe/circe-core_3/0.14.11/build.gradle
@@ -1,0 +1,36 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+plugins {
+    id "org.graalvm.internal.tck"
+    id "scala"
+}
+
+String libraryVersion = tck.testedLibraryVersion.get()
+
+dependencies {
+    testImplementation "io.circe:circe-core_3:$libraryVersion"
+    testImplementation 'org.assertj:assertj-core:3.22.0'
+    testImplementation 'org.scala-lang:scala3-library_3:3.3.6'
+    testImplementation 'org.scala-lang:scala3-compiler_3:3.3.6'
+}
+
+java {
+    toolchain {
+        languageVersion = JavaLanguageVersion.of(21)
+    }
+}
+
+graalvmNative {
+    agent {
+        defaultMode = "conditional"
+        modes {
+            conditional {
+                userCodeFilterPath = "user-code-filter.json"
+            }
+        }
+    }
+}

--- a/tests/src/io.circe/circe-core_3/0.14.11/gradle.properties
+++ b/tests/src/io.circe/circe-core_3/0.14.11/gradle.properties
@@ -1,0 +1,6 @@
+# Optional explicit tested library coordinates (format: group:artifact:version).
+# If omitted, fallback resolution in org.graalvm.internal.tck.gradle still uses
+# environment variable GVM_TCK_LC and then this property.
+library.coordinates = io.circe:circe-core_3:0.14.11
+library.version = 0.14.11
+metadata.dir = io.circe/circe-core_3/0.14.11/

--- a/tests/src/io.circe/circe-core_3/0.14.11/settings.gradle
+++ b/tests/src/io.circe/circe-core_3/0.14.11/settings.gradle
@@ -1,0 +1,20 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+pluginManagement {
+    def tckPath = Objects.requireNonNullElse(
+            System.getenv("GVM_TCK_TCKDIR"),
+            "../../../../tck-build-logic"
+    )
+    includeBuild(tckPath)
+    repositories { mavenLocal(); gradlePluginPortal() }
+}
+
+plugins {
+    id "org.graalvm.internal.tck-settings" version "1.0.0-SNAPSHOT"
+}
+
+rootProject.name = 'io.circe.circe-core_3_tests'

--- a/tests/src/io.circe/circe-core_3/0.14.11/src/test/scala/io_circe/circe_core_3/Circe_core_3Test.scala
+++ b/tests/src/io.circe/circe-core_3/0.14.11/src/test/scala/io_circe/circe_core_3/Circe_core_3Test.scala
@@ -16,9 +16,6 @@ import io.circe.JsonObject
 import io.circe.KeyDecoder
 import io.circe.KeyEncoder
 import io.circe.Printer
-import io.circe.derivation.ConfiguredCodec
-import io.circe.derivation.ConfiguredEnumCodec
-import io.circe.derivation.Configuration
 import io.circe.syntax._
 import org.assertj.core.api.Assertions.assertThat
 import org.junit.jupiter.api.Assertions.fail
@@ -59,12 +56,49 @@ object AccountId {
 
 final case class Command(action: String, target: Option[String])
 
-given Configuration = Configuration.default.withSnakeCaseMemberNames.withDefaults.withStrictDecoding
+final case class DerivedJob(jobId: String, ownerName: String, maxRetries: Int = 3)
 
-final case class DerivedJob(jobId: String, ownerName: String, maxRetries: Int = 3) derives ConfiguredCodec
+object DerivedJob {
+  private val jsonFields: Set[String] = Set("job_id", "owner_name", "max_retries")
 
-enum TicketState derives ConfiguredEnumCodec {
+  given Encoder.AsObject[DerivedJob] = Encoder.AsObject.instance { job =>
+    JsonObject(
+      "job_id" -> Json.fromString(job.jobId),
+      "owner_name" -> Json.fromString(job.ownerName),
+      "max_retries" -> Json.fromInt(job.maxRetries)
+    )
+  }
+
+  given Decoder[DerivedJob] = Decoder.instance { cursor =>
+    cursor.keys match {
+      case Some(keys) =>
+        val unexpectedFields: List[String] = keys.filterNot(field => jsonFields.contains(field)).toList
+        if unexpectedFields.nonEmpty then
+          Left(DecodingFailure(s"Unexpected fields: ${unexpectedFields.mkString(", ")}", cursor.history))
+        else
+          for {
+            jobId <- cursor.get[String]("job_id")
+            ownerName <- cursor.get[String]("owner_name")
+            maxRetries <- cursor.get[Option[Int]]("max_retries").map(_.getOrElse(3))
+          } yield DerivedJob(jobId, ownerName, maxRetries)
+      case None => Left(DecodingFailure("Expected JSON object", cursor.history))
+    }
+  }
+}
+
+enum TicketState {
   case Open, InProgress, Closed
+}
+
+object TicketState {
+  given Encoder[TicketState] = Encoder.instance(state => Json.fromString(state.toString))
+
+  given Decoder[TicketState] = Decoder.decodeString.emap {
+    case "Open" => Right(TicketState.Open)
+    case "InProgress" => Right(TicketState.InProgress)
+    case "Closed" => Right(TicketState.Closed)
+    case other => Left(s"enum TicketState does not contain case: $other")
+  }
 }
 
 object Command {
@@ -240,7 +274,7 @@ class Circe_core_3Test {
   }
 
   @Test
-  def derivesConfiguredCodecsForProductsWithDefaultsAndStrictDecoding(): Unit = {
+  def encodesAndDecodesProductsWithDefaultsAndStrictDecoding(): Unit = {
     val encoded: Json = DerivedJob("job-1", "Ada", 5).asJson
     assertThat(encoded.noSpacesSortKeys).isEqualTo("""{"job_id":"job-1","max_retries":5,"owner_name":"Ada"}""")
 
@@ -259,7 +293,7 @@ class Circe_core_3Test {
   }
 
   @Test
-  def derivesCodecsForSingletonEnums(): Unit = {
+  def encodesAndDecodesSingletonEnums(): Unit = {
     assertThat(TicketState.Open.asJson).isEqualTo(Json.fromString("Open"))
     assertThat(TicketState.InProgress.asJson).isEqualTo(Json.fromString("InProgress"))
     assertThat(Json.fromString("Closed").as[TicketState]).isEqualTo(Right(TicketState.Closed))

--- a/tests/src/io.circe/circe-core_3/0.14.11/src/test/scala/io_circe/circe_core_3/Circe_core_3Test.scala
+++ b/tests/src/io.circe/circe-core_3/0.14.11/src/test/scala/io_circe/circe_core_3/Circe_core_3Test.scala
@@ -1,0 +1,293 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package io_circe.circe_core_3
+
+import io.circe.Decoder
+import io.circe.DecodingFailure
+import io.circe.Encoder
+import io.circe.HCursor
+import io.circe.Json
+import io.circe.JsonNumber
+import io.circe.JsonObject
+import io.circe.KeyDecoder
+import io.circe.KeyEncoder
+import io.circe.Printer
+import io.circe.derivation.ConfiguredCodec
+import io.circe.derivation.ConfiguredEnumCodec
+import io.circe.derivation.Configuration
+import io.circe.syntax._
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Assertions.fail
+import org.junit.jupiter.api.Test
+
+import java.nio.ByteBuffer
+import java.nio.charset.StandardCharsets
+import java.time.Instant
+import java.util.UUID
+import scala.jdk.CollectionConverters._
+import scala.util.Try
+
+final case class Profile(
+    name: String,
+    age: Int,
+    created: Instant,
+    aliases: List[String],
+    metadata: Map[UUID, BigDecimal]
+)
+
+object Profile {
+  given Encoder[Profile] = Encoder.forProduct5("name", "age", "created", "aliases", "metadata") { profile =>
+    (profile.name, profile.age, profile.created, profile.aliases, profile.metadata)
+  }
+
+  given Decoder[Profile] = Decoder.forProduct5("name", "age", "created", "aliases", "metadata")(Profile.apply)
+}
+
+final case class AccountId(value: UUID)
+
+object AccountId {
+  given KeyEncoder[AccountId] = KeyEncoder.instance(_.value.toString)
+
+  given KeyDecoder[AccountId] = KeyDecoder.instance { key =>
+    Try(UUID.fromString(key)).toOption.map(AccountId.apply)
+  }
+}
+
+final case class Command(action: String, target: Option[String])
+
+given Configuration = Configuration.default.withSnakeCaseMemberNames.withDefaults.withStrictDecoding
+
+final case class DerivedJob(jobId: String, ownerName: String, maxRetries: Int = 3) derives ConfiguredCodec
+
+enum TicketState derives ConfiguredEnumCodec {
+  case Open, InProgress, Closed
+}
+
+object Command {
+  given Decoder[Command] = Decoder.instance { (cursor: HCursor) =>
+    val actionCursor = cursor.downField("action")
+    cursor.get[String]("action").flatMap {
+      case "start" => cursor.get[String]("target").map(target => Command("start", Some(target)))
+      case "stop" => Right(Command("stop", None))
+      case other => Left(DecodingFailure(s"Unsupported action: $other", actionCursor.history))
+    }
+  }
+}
+
+class Circe_core_3Test {
+  @Test
+  def buildsTransformsAndPrintsJsonDocuments(): Unit = {
+    val document: Json = Json.obj(
+      "b" -> Json.Null,
+      "a" -> Json.arr(Json.fromInt(2), Json.fromString("text")),
+      "nested" -> Json.obj("keep" -> Json.True, "drop" -> Json.Null)
+    )
+
+    assertThat(document.isObject).isTrue
+    assertThat(document.hcursor.downField("nested").downField("keep").as[Boolean]).isEqualTo(Right(true))
+    assertThat(Json.fromDouble(Double.NaN).isEmpty).isTrue
+    assertThat(Json.fromDoubleOrString(Double.NaN).asString).isEqualTo(Some("NaN"))
+
+    val transformed: Json = document.deepDropNullValues.mapObject(_.add("z", Json.fromLong(3L)))
+    assertThat(transformed.noSpacesSortKeys).isEqualTo("""{"a":[2,"text"],"nested":{"keep":true},"z":3}""")
+
+    val pretty: String = Printer.spaces2SortKeys.print(transformed)
+    assertThat(pretty).contains("\n  \"a\" : [")
+    assertThat(pretty).contains("\n  \"nested\" : {")
+
+    val compactBuffer: ByteBuffer = Printer.noSpaces.printToByteBuffer(transformed, StandardCharsets.UTF_8)
+    val compactBytes: Array[Byte] = new Array[Byte](compactBuffer.remaining)
+    compactBuffer.get(compactBytes)
+    assertThat(new String(compactBytes, StandardCharsets.UTF_8)).isEqualTo(transformed.noSpaces)
+  }
+
+  @Test
+  def manipulatesJsonObjectsWithoutLosingInsertionOrder(): Unit = {
+    val base: JsonObject = JsonObject(
+      "name" -> Json.fromString("Ada"),
+      "nullable" -> Json.Null
+    )
+    val updated: JsonObject = base
+      .add("enabled", Json.fromBoolean(true))
+      .remove("nullable")
+      .mapValues(_.withString(value => Json.fromString(value.toUpperCase)))
+
+    assertThat(updated.keys.toSeq.asJava).containsExactly("name", "enabled")
+    assertThat(updated("name")).isEqualTo(Some(Json.fromString("ADA")))
+    assertThat(updated.contains("nullable")).isFalse
+    assertThat(updated.size).isEqualTo(2)
+
+    val left: JsonObject = JsonObject(
+      "config" -> Json.obj("threads" -> Json.fromInt(2), "debug" -> Json.False),
+      "owner" -> Json.fromString("test")
+    )
+    val right: JsonObject = JsonObject(
+      "config" -> Json.obj("debug" -> Json.True, "region" -> Json.fromString("eu"))
+    )
+
+    assertThat(left.deepMerge(right).toJson.noSpacesSortKeys)
+      .isEqualTo("""{"config":{"debug":true,"region":"eu","threads":2},"owner":"test"}""")
+  }
+
+  @Test
+  def navigatesEditsAndDeletesWithCursors(): Unit = {
+    val document: Json = Json.obj(
+      "users" -> Json.arr(
+        Json.obj("name" -> Json.fromString("Ada"), "scores" -> Json.arr(Json.fromInt(1), Json.fromInt(2))),
+        Json.obj("name" -> Json.fromString("Linus"), "scores" -> Json.arr(Json.fromInt(3), Json.fromInt(5)))
+      )
+    )
+
+    val secondName = document.hcursor.downField("users").downN(1).downField("name")
+    assertThat(secondName.succeeded).isTrue
+    assertThat(secondName.key).isEqualTo(Some("name"))
+    assertThat(secondName.as[String]).isEqualTo(Right("Linus"))
+    assertThat(secondName.pathString).isEqualTo(".users[1].name")
+
+    val incremented: Json = decodeOptionOrFail(
+      document.hcursor
+        .downField("users")
+        .downArray
+        .downField("scores")
+        .downN(1)
+        .withFocus(incrementJsonIntegerBy(10))
+        .top
+    )
+    assertThat(incremented.hcursor.downField("users").downArray.downField("scores").downN(1).as[Int])
+      .isEqualTo(Right(12))
+
+    val withoutFirstUserScores: Json = decodeOptionOrFail(
+      incremented.hcursor.downField("users").downArray.downField("scores").delete.top
+    )
+    assertThat(withoutFirstUserScores.hcursor.downField("users").downArray.downField("scores").succeeded).isFalse
+    assertThat(withoutFirstUserScores.hcursor.downField("users").downN(1).downField("scores").as[List[Int]])
+      .isEqualTo(Right(List(3, 5)))
+  }
+
+  @Test
+  def encodesAndDecodesProductsCollectionsAndJavaTimeValues(): Unit = {
+    val firstMetadataKey: UUID = UUID.fromString("00000000-0000-0000-0000-000000000001")
+    val secondMetadataKey: UUID = UUID.fromString("00000000-0000-0000-0000-000000000002")
+    val profile: Profile = Profile(
+      name = "Ada Lovelace",
+      age = 36,
+      created = Instant.parse("2024-01-02T03:04:05Z"),
+      aliases = List("analyst", "programmer"),
+      metadata = Map(firstMetadataKey -> BigDecimal("12.5"), secondMetadataKey -> BigDecimal("7"))
+    )
+
+    val json: Json = profile.asJson
+    assertThat(json.hcursor.get[String]("name")).isEqualTo(Right("Ada Lovelace"))
+    assertThat(json.hcursor.get[Instant]("created")).isEqualTo(Right(profile.created))
+    assertThat(decodeOrFail(json.hcursor.get[List[String]]("aliases")).asJava).containsExactly("analyst", "programmer")
+    assertThat(decodeOrFail(json.hcursor.get[Map[UUID, BigDecimal]]("metadata"))).isEqualTo(profile.metadata)
+    assertThat(json.as[Profile]).isEqualTo(Right(profile))
+  }
+
+  @Test
+  def accumulatesFieldDecodingFailuresForProductDecoders(): Unit = {
+    val invalidProfile: Json = Json.obj(
+      "name" -> Json.fromInt(42),
+      "age" -> Json.fromString("old"),
+      "created" -> Json.fromString("not-an-instant"),
+      "aliases" -> Json.fromString("not-a-list"),
+      "metadata" -> Json.obj("not-a-uuid" -> Json.fromString("not-a-number"))
+    )
+
+    val failures: List[DecodingFailure] = invalidProfile.asAccumulating[Profile].fold(
+      _.toList,
+      value => fail(s"Expected decoding to fail, but decoded: $value")
+    )
+
+    assertThat(failures.size).isGreaterThanOrEqualTo(4)
+    assertThat(failures.map(_.history.nonEmpty).asJava).contains(true)
+  }
+
+  @Test
+  def encodesAndDecodesMapsWithCustomKeyCodecs(): Unit = {
+    val first: AccountId = AccountId(UUID.fromString("11111111-1111-1111-1111-111111111111"))
+    val second: AccountId = AccountId(UUID.fromString("22222222-2222-2222-2222-222222222222"))
+
+    val json: Json = Map(first -> 10, second -> 20).asJson
+
+    assertThat(json.noSpacesSortKeys)
+      .isEqualTo("""{"11111111-1111-1111-1111-111111111111":10,"22222222-2222-2222-2222-222222222222":20}""")
+    assertThat(json.as[Map[AccountId, Int]]).isEqualTo(Right(Map(first -> 10, second -> 20)))
+
+    val invalid: Either[DecodingFailure, Map[AccountId, Int]] = Json.obj("not-a-uuid" -> Json.fromInt(1)).as[Map[AccountId, Int]]
+    assertThat(invalid.isLeft).isTrue
+  }
+
+  @Test
+  def usesCustomDecoderInstancesForDomainValidation(): Unit = {
+    val start: Json = Json.obj("action" -> Json.fromString("start"), "target" -> Json.fromString("service-a"))
+    val stop: Json = Json.obj("action" -> Json.fromString("stop"))
+    val unsupported: Json = Json.obj("action" -> Json.fromString("restart"), "target" -> Json.fromString("service-a"))
+
+    assertThat(start.as[Command]).isEqualTo(Right(Command("start", Some("service-a"))))
+    assertThat(stop.as[Command]).isEqualTo(Right(Command("stop", None)))
+
+    val error: DecodingFailure = unsupported.as[Command] match {
+      case Left(error) => error
+      case Right(value) => fail(s"Expected unsupported action to fail, but decoded: $value")
+    }
+    assertThat(error.message).isEqualTo("Unsupported action: restart")
+    assertThat(error.history.asJava).isNotEmpty
+  }
+
+  @Test
+  def derivesConfiguredCodecsForProductsWithDefaultsAndStrictDecoding(): Unit = {
+    val encoded: Json = DerivedJob("job-1", "Ada", 5).asJson
+    assertThat(encoded.noSpacesSortKeys).isEqualTo("""{"job_id":"job-1","max_retries":5,"owner_name":"Ada"}""")
+
+    val decodedWithDefault: Either[DecodingFailure, DerivedJob] = Json.obj(
+      "job_id" -> Json.fromString("job-2"),
+      "owner_name" -> Json.fromString("Linus")
+    ).as[DerivedJob]
+    assertThat(decodedWithDefault).isEqualTo(Right(DerivedJob("job-2", "Linus")))
+
+    val decodedWithUnknownField: Either[DecodingFailure, DerivedJob] = Json.obj(
+      "job_id" -> Json.fromString("job-3"),
+      "owner_name" -> Json.fromString("Grace"),
+      "unexpected" -> Json.True
+    ).as[DerivedJob]
+    assertThat(decodedWithUnknownField.isLeft).isTrue
+  }
+
+  @Test
+  def derivesCodecsForSingletonEnums(): Unit = {
+    assertThat(TicketState.Open.asJson).isEqualTo(Json.fromString("Open"))
+    assertThat(TicketState.InProgress.asJson).isEqualTo(Json.fromString("InProgress"))
+    assertThat(Json.fromString("Closed").as[TicketState]).isEqualTo(Right(TicketState.Closed))
+
+    val invalid: Either[DecodingFailure, TicketState] = Json.fromString("Blocked").as[TicketState]
+    assertThat(invalid.isLeft).isTrue
+  }
+
+  @Test
+  def preservesJsonNumberPrecisionAndRejectsInvalidNumericInput(): Unit = {
+    val number: JsonNumber = decodeOptionOrFail(JsonNumber.fromString("12345678901234567890.125"))
+    val json: Json = Json.fromJsonNumber(number)
+
+    assertThat(json.asNumber.flatMap(_.toBigDecimal)).isEqualTo(Some(BigDecimal("12345678901234567890.125")))
+    assertThat(json.as[BigDecimal]).isEqualTo(Right(BigDecimal("12345678901234567890.125")))
+    assertThat(JsonNumber.fromString("1.2.3").isEmpty).isTrue
+    assertThat(JsonNumber.fromString("not-a-number").isEmpty).isTrue
+  }
+
+  private def incrementJsonIntegerBy(amount: Int)(json: Json): Json = {
+    json.asNumber.flatMap(_.toInt).map(value => Json.fromInt(value + amount)).getOrElse(json)
+  }
+
+  private def decodeOrFail[A](result: Either[DecodingFailure, A]): A = {
+    result.fold(error => fail(s"Expected decoding success but got: $error"), identity)
+  }
+
+  private def decodeOptionOrFail[A](option: Option[A]): A = {
+    option.getOrElse(fail("Expected value to be present"))
+  }
+}

--- a/tests/src/io.circe/circe-core_3/0.14.11/user-code-filter.json
+++ b/tests/src/io.circe/circe-core_3/0.14.11/user-code-filter.json
@@ -1,0 +1,13 @@
+{
+  "rules" : [
+    {
+      "excludeClasses" : "**"
+    },
+    {
+      "includeClasses" : "io.circe.**"
+    },
+    {
+      "includeClasses" : "io_circe.circe_core_3.**"
+    }
+  ]
+}

